src/main/java/fi/aalto/cs/apluscourses/utils/observable/ObservableProperty.java

```diff
@@ -67,4 +90,29 @@ default void valueChangedUntyped(@NotNull Object observer, @Nullable T value) {
       valueChanged((O) observer, value);
     }
   }
+
+  public interface SimpleCallback<O> {
+    void valueChanged(@NotNull O observer);
+  }
+
+  private static class CallbackWrapper<O, T> implements Callback<O, T> {
+
+    @NotNull
+    private final SimpleCallback<O> callback;
+
+    public CallbackWrapper(@NotNull SimpleCallback<O> callback) {
+      this.callback = callback;
+    }
+
+    @Override
+    public void valueChanged(@NotNull O observer, @Nullable T value) {
+      callback.valueChanged(observer);
+    }
+  }
+
+  @FunctionalInterface
+  public static interface Validator<T> {
```

Comment:
"Modifier 'static' is redundant for inner interfaces"

Reply:
Apparently, yes. 👍